packages/es-components/src/components/containers/loading-skeleton/LoadingSkeleton.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { ThemeProvider } from 'styled-components';
+import tinycolor from 'tinycolor2';
+import { useTheme } from '../../util/useTheme';
+
+const SkeletonContainer = styled.div`
+  position: relative;
+  background-color: ${({ theme: { skeleton } }) => skeleton.shimmerColor};
+
+  &::after {
+    animation: loading-keyframes 1.5s linear infinite;
+    background-size: 50%;
+    background-repeat: no-repeat;
+    background-image: linear-gradient(
+      -60deg,
+      ${({ theme: { skeleton } }) =>
+          tinycolor(skeleton.shimmerColor)
+            .setAlpha(0)
+            .toRgbString()}
+        0,
+      ${({ theme: { skeleton } }) =>
+          tinycolor(skeleton.shimmerColor)
+            .setAlpha(0)
+            .toRgbString()}
+        40%,
+      ${({ theme: { skeleton } }) =>
+          tinycolor(skeleton.shimmerColor)
+            .setAlpha(0.8)
+            .toRgbString()}
+        50%,
+      ${({ theme: { skeleton } }) =>
+          tinycolor(skeleton.shimmerColor)
+            .setAlpha(0)
+            .toRgbString()}
+        60%,
+      ${({ theme: { skeleton } }) =>
+          tinycolor(skeleton.shimmerColor)
+            .setAlpha(0)
+            .toRgbString()}
+        100%
+    );
+    content: '';
+    display: block;
+    height: 100%;
+    left: 0;
+    position: absolute;
+    top: 0;
+    width: 100%;
+  }
+
+  @keyframes loading-keyframes {
+    from {
+      background-position: -150% 0;
+    }
+    to {
+      background-position: 250% 0;
+    }
+  }
+`;
+
+const SkeletonShape = styled.div`
+  background-color: ${({ theme: { skeleton } }) => skeleton.shapeColor};
+`;
+
+const LoadingSkeleton = ({ shapeColor, shimmerColor, children, ...props }) => {
+  const theme = useTheme();
+  const { colors } = theme;
+  const shimmer = shimmerColor || colors.white;
+  const shape = shapeColor || colors.gray3;
+  const skeleton = { shimmerColor: shimmer, shapeColor: shape };
+  return (
+    <ThemeProvider theme={{ ...theme, skeleton }}>
```

Comment:
Spreading `theme` is unnecessary. `ThemeProvider` [does this for you](https://github.com/styled-components/styled-components/blob/271e70e6203f70ce91f06bca634d7040ade14964/packages/styled-components/src/models/ThemeProvider.js#L47-L59). See `mergeTheme` for details.

Reply:
I was wondering about this! I'll update!